packages/clerk-js/src/v4/customizables/elementDescriptors.ts

```diff
@@ -0,0 +1,87 @@
+import { ElementObjectKey, ElementsConfig, IdSelectors, StateSelectors } from '@clerk/types';
+
+export const CLASS_PREFIX = 'cl-';
+export const ID_CLASS_PREFIX = 'cl-id-';
+export const OBJ_KEY_DELIMITER = '__';
+
+/**
+ * This object is strictly typed using the ElementsConfig type
+ * and is used as the single source of truth to generate the
+ * descriptor map
+ */
+export const APPEARANCE_KEYS: Readonly<Array<keyof ElementsConfig>> = [
```

Comment:
This needs to be kept in sync with the `ElementConfig` type manually. This is by design - `ElementsConfig` is the single source of truth, so `APPEARANCE_KEYS` can't change unless we make a change to `ElementsConfig` as well